AdyenEncryption/BankDetailsEncryptor.swift

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright (c) 2022 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// A type that provides static methods for encrypting bank details.
+public enum BankDetailsEncryptor: AnyEncryptor {
+    
+    /// Encrypts bank account number.
+    ///
+    /// - Parameters:
+    ///   - number: The bank account number
+    ///   - publicKey: The public key to use for encryption (format "Exponent|Modulus").
+    /// - Returns: An encrypted token containing card number data.
+    /// - Throws: `CardEncryptor.Error.encryptionFailed` if the encryption failed,
+    ///  maybe because the card public key is an invalid one, or for any other reason.
+    /// - Throws: `CardEncryptor.Error.emptyValue` when trying to encrypt an empty or invalid card number
+    public static func encrypt(accountNumber: String, with publicKey: String) throws -> String {
+        guard !accountNumber.isEmpty, accountNumber.allSatisfy(\.isNumber) else {
+            throw CardEncryptor.Error.emptyValue
+        }
+        let payload = BankPayload().add(accountNumber: accountNumber)
+        return try encrypt(payload, with: publicKey)
+    }
+    
+    /// Encrypts bank routing number.
+    ///
+    /// - Parameters:
+    ///   - number: The bank account number
+    ///   - publicKey: The public key to use for encryption (format "Exponent|Modulus").
+    /// - Returns: An encrypted token containing card number data.
+    /// - Throws: `CardEncryptor.Error.encryptionFailed` if the encryption failed,
+    ///  maybe because the card public key is an invalid one, or for any other reason.
+    /// - Throws: `CardEncryptor.Error.emptyValue` when trying to encrypt an empty or invalid card number
+    public static func encrypt(routingNumber: String, with publicKey: String) throws -> String {
+        guard !routingNumber.isEmpty, routingNumber.allSatisfy(\.isNumber) else {
+            throw CardEncryptor.Error.emptyValue
```

Comment:
maywe we can change the name of the error to `invalidValue`, same in line 43

Reply:
They are public unfortunately.

Reply:
we probably want to have a different error enum then? since this is not a `CardEnryptor,` but a `BankDetailsEncryptor`

Reply:
This one is tricky since that enum contains some generic errors too like `encryptionFailed` which both could use. 
I did not want to create a new enum under this class only for the new cases. Wasn't ideal, the best would be to modify this but it is public.

Reply:
Added a new enum case in the end.